plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -235,6 +269,15 @@ protected function prepare_objects_query( $request ) {
 			$args['meta_key'] = $ordering_args['meta_key']; // WPCS: slow query ok.
 		}
 
+		/*
+		 * When the related products ids is not empty,
+		 * filter the query to return only the related products,
+		 * overwriting the post__in parameter.
+		 */
+		if ( ! empty( $this->related_producs_ids ) ) {
+			$args['post__in'] = $this->related_producs_ids;
```

Comment:
Aaah I was curious how the `parent::get_items( $request );` was still working, but I guess that is because of this.

Do you think it would make more sense if we just call `wc_get_product` here with array_map, I saw something similar in one of the template functions ( don't have to copy this, just an example of array_map:
```
$args['related_products'] = array_filter( array_map( 'wc_get_product', wc_get_related_products( $product->get_id(), $args['posts_per_page'], $product->get_upsell_ids() ) ), 'wc_products_array_filter_visible' );
// Handle orderby.
$args['related_products'] = wc_products_array_orderby( $args['related_products'], $args['orderby'], $args['order'] );
```

Reply:
The (good, I think) thing about using the parent get_items() is that it will inherit the whole behavior. For instance, could I use the `_fields`, `per_page,` and params? I think we could, but it will require addressing not only this one but the rest of them: per_page, etc.

Reply:
Do you see any issue by using the get_items() method? In the end, what we need to return is a products collection